dmriprep/workflows/dwi/util.py

```diff
@@ -292,3 +292,178 @@ def _normalize(in_file, newmax=2000, perc=98.0):
+
+    # Generate output file name
+    out_file = fname_presuffix(
+        in_file,
+        suffix="_acqparams.txt",
+        use_ext=False,
+    )
+
+    # Dictionary for converting directions to acqp lines
+    acq_param_dict = {
+        "j": "0 1 0 %.7f",
+        "j-": "0 -1 0 %.7f",
+        "i": "1 0 0 %.7f",
+        "i-": "-1 0 0 %.7f",
+        "k": "0 0 1 %.7f",
+        "k-": "0 0 -1 %.7f",
+    }
+
+    # Get encoding direction from json
+    if metadata.get("PhaseEncodingDirection"):
+        pe_dir = metadata.get("PhaseEncodingDirection")
+    else:
+        pe_dir = metadata.get("PhaseEncodingAxis")
+
+    # Get readout time from json, use default of 0.05 otherwise
+    if metadata.get("TotalReadoutTime"):
+        total_readout = metadata.get("TotalReadoutTime")
+    else:
+        total_readout = total_readout_time
+
+    # Construct the acqp file lines
+    acq_param_lines = acq_param_dict[pe_dir] % total_readout
+
+    # Write to the acqp file
+    Path(out_file).write_text(acq_param_lines)
+
+    return out_file
+
+
+def init_eddy_wf(name="eddy_wf"):
+    """
+    Create a workflow for head-motion & Eddy currents distortion estimation with FSL ``eddy``.
+
+    Parameters
+    ----------
+    name : :obj:`str`
+        Name of workflow (default: ``eddy_wf``)
+
+    Inputs
+    ----------
+    dwi_file
+        dwi NIfTI file
+    dwi_mask
+        Skull-stripping mask of reference image
+    in_bvec
+        File containing bvecs of dwi
+    in_bval
+        File containing bvals of dwi
+
+    Outputs
+    -------
+    out_eddy :
+        The eddy corrected diffusion image.
+    out_rotated_bvecs :
+        Rotated bvecs for each volume after eddy.
+
+    """
+    from nipype.interfaces.fsl import Eddy, EddyQuad
+
+    inputnode = pe.Node(
+        niu.IdentityInterface(
+            fields=[
+                "dwi_file",
+                "dwi_mask",
+                "in_bvec",
+                "in_bval",
+            ]
+        ),
+        name="inputnode",
+    )
+
+    outputnode = pe.Node(
+        niu.IdentityInterface(
+            fields=[
+                "out_eddy",
+                "out_rotated_bvecs"
+            ]
+        ),
+        name="outputnode",
+    )
+
+    workflow = Workflow(name=name)
+
+    eddy = pe.Node(
+        Eddy(repol=True, cnr_maps=True, residuals=True, method="jac"),
+        name="eddy",
+    )
+
+    # Generate the acqp file for eddy
+    eddy_acqp = pe.Node(
+        niu.Function(
+            input_names=["in_file"],
+            output_names=["out_file"],
+            function=gen_acqparams,
+        ),
+        name="eddy_acqp",
+    )
+
+    # Generate the index file for eddy
+    gen_idx = pe.Node(
+        niu.Function(
+            input_names=["in_file"],
+            output_names=["out_file"],
+            function=gen_index,
+        ),
+        name="gen_idx",
+    )
+
+    eddy_quad = pe.Node(EddyQuad(verbose=True), name="eddy_quad")
```

Comment:
We won't need to include `eddy_quad`

Reply:
Thanks for the catch, fixed in 6605fe19d919b72c392a79d6c9f3e518c076685d